libkbfs/block_retrieval_queue.go

```diff
@@ -74,18 +78,38 @@ func newBlockRetrievalQueue(numWorkers int) *blockRetrievalQueue {
 // notifyWorker notifies workers that there is a new request for processing.
 func (brq *blockRetrievalQueue) notifyWorker() {
 	go func() {
-		// Get the next queued worker
-		ch := <-brq.workerQueue
-		// Prevent interference with the heap while we're retrieving from it
-		brq.mtx.Lock()
-		defer brq.mtx.Unlock()
-		// Pop from the heap
-		ch <- heap.Pop(brq.heap).(*blockRetrieval)
+		select {
+		case <-brq.doneCh:
+			// Prevent interference with the heap while we're retrieving from it
+			brq.mtx.Lock()
+			defer brq.mtx.Unlock()
+			if brq.heap.Len() > 0 {
+				retrieval := heap.Pop(brq.heap).(*blockRetrieval)
+				brq.FinalizeRequest(retrieval, nil, io.EOF)
+			}
+		default:
+			// Get the next queued worker
```

Comment:
You can avoid indentation by returning above, moving this block out of the `select` block, and just have the `default` case be empty.  That seems a little clearer to me, but I don't feel too strongly about it.